src/middlewared/middlewared/etc_files/sysctl_config.py

```diff
@@ -1,20 +1,49 @@
 import subprocess
+import sysctl
+
+default_sysctl = {}
 
 
 def sysctl_configuration(middleware):
-    tuneables = middleware.call_sync('tunable.query', [
-        ['enabled', '=', True], ['type', '=', 'SYSCTL']
-    ])
+    tuneables = middleware.call_sync('tunable.query', [['type', '=', 'SYSCTL']])
     for tuneable in tuneables:
-        ret = subprocess.run(
-            ['sysctl', f'{tuneable["var"]}="{tuneable["value"]}"'],
-            capture_output=True
-        )
-        if ret.returncode:
-            middleware.logger.debug(
-                'Failed to set sysctl %s -> %s: %s',
-                tuneable['var'], tuneable['value'], ret.stderr.decode(),
+        value_default = None
+        if tuneable['type'].lower() == 'sysctl':
+            try:
+                value_default = middleware.call_sync('tunable.get_default_value', tuneable['var'])
+            except KeyError:
+                pass
+        if tuneable['enabled'] is True:
+            if value_default is None and tuneable['type'].lower() == 'sysctl':
```

Comment:
Why are you checking for type if you are already filtering for type in the query?